presentation/src/main/java/dk/nodes/template/presentation/ui/base/BaseFragment.kt

```diff
@@ -14,4 +14,9 @@ abstract class BaseFragment : DaggerFragment() {
         ViewModelProviders.of(this, viewModelFactory)
             .get(VM::class.java)
     }
+
+    protected inline fun <reified VM : ViewModel> sharedViewModel(): Lazy<VM> = lazy {
```

Comment:
/**
 * Returns a shared ViewModel which can be used by different child Fragments
 * */
protected inline fun <reified VM : ViewModel> bindParentViewModel(): VM {
    val provider = parentFragment?.let { ViewModelProviders.of(it, viewModelFactory) } 
            ?: activity?.let { ViewModelProviders.of(it, viewModelFactory) }
    return provider?.get(VM::class.java) ?: throw Exception("Failed to find parent ViewModel in ${this::class.simpleName}")
}